src/resources/2010/LookingGlass.ts

```diff
@@ -0,0 +1,99 @@
+import { get } from "../../property";
+import { $effect, $slot } from "../../template-string";
+import { ensureEffect, have } from "../../lib";
+import { canEquip, cliExecute, haveEffect, toInt, toSlot } from "kolmafia";
+
+const firstTeaPartyEffect = $effect`Assaulted with Pepper`;
+const lastTeaPartyEffect = $effect`Oleaginous Soles`;
+
+/**
+ * Checks if an item is a valid hat for The Mad Tea Party
+ * @param item The item to check
+ * @param characters The desired nameLength for The Mad Tea Party effect
+ * @returns If this item is a valid hat for The Mad Tea Party
+ */
+function validTeaPartyHat(item: Item, characters: number): boolean {
+  if (toSlot(item) !== $slot`hat` || !canEquip(item)) {
+    return false;
+  } else if (characters <= 5) {
+    return item.nameLength <= 5;
+  } else if (characters >= 30) {
+    return item.nameLength >= 30;
+  }
+  return characters === item.nameLength;
+}
+
+/**
+ * Find the name length required to get an effect from The Mad Tea Party
+ * @param effect The effect to check
+ * @returns The number of characters required to get an effect
+ */
+export function findTeaPartyHatLength(effect: Effect): number {
+  const effectId = toInt(effect);
+  if (
+    effectId < toInt(firstTeaPartyEffect) ||
+    effectId > toInt(lastTeaPartyEffect)
+  ) {
+    throw `Invalid Mad Tea Party effect ${effect}`;
+  }
+  return Math.floor(effectId - toInt(firstTeaPartyEffect) + 5);
+}
+
+/**
+ * Find all hats that will get an effect from The Mad Tea Party
+ * @param effect The effect to check
+ * @returns An array of equippable hats
+ */
+export function findTeaPartyHats(effect: Effect): Item[] {
+  const characters = findTeaPartyHatLength(effect);
+  if (!characters) {
+    throw `Invalid Mad Tea Party effect ${effect}`;
+  }
+  return Item.all().filter((item) => validTeaPartyHat(item, characters));
+}
+
+/**
+ * Check if a hat is available to get an effect from The Mad Tea Party
+ * @param effect The effect to check
+ * @returns If any valid hat is in inventory
+ */
+export function haveTeaPartyHat(effect: Effect): boolean {
+  const characters = findTeaPartyHatLength(effect);
+  if (!characters) {
+    throw `Invalid Mad Tea Party effect ${effect}`;
+  }
+  return (
+    Item.all().find(
+      (item) => validTeaPartyHat(item, characters) && have(item)
```

Comment:
you need to be able to equip it too, not just have it

Reply:
It's checked in validTeaPartyHat:
```/**
 * Checks if an item is a valid hat for The Mad Tea Party
 * @param item The item to check
 * @param characters The desired nameLength for The Mad Tea Party effect
 * @returns If this item is a valid hat for The Mad Tea Party
 */
function validTeaPartyHat(item: Item, characters: number): boolean {
  if (toSlot(item) !== $slot`hat` || !canEquip(item)) {
    return false;
  } else if (characters <= 5) {
    return item.nameLength <= 5;
  } else if (characters >= 30) {
    return item.nameLength >= 30;
  }
  return characters === item.nameLength;
}
```

Reply:
thanks, sorry